internal/cmd/io_binary.go

```diff
@@ -105,8 +115,24 @@ func NewBinaryIO(ctx context.Context, id string, uri *url.URL) (_ UpstreamIO, er
 	}, nil
 }
 
+// sanitizePath parses the URL object and returns a clean path to the logging driver
+func sanitizePath(uri *url.URL) string {
+	path := strings.Replace(uri.Path, "/", `\`, -1)
+	path = filepath.Clean(path)
+
+	if path == "." {
+		return ""
+	}
+
+	if strings.Contains(path, `:\`) {
```

Comment:
Can we just remove these two `if` blocks? What are they needed for?

Reply:
the first `if` can probably go, since if it ends up being an empty path, then it's a client error and it was more important when we handled `uri.Host`.

in the second one I'm trying to handle these cases:
`/path/on/default/volume` (or `/path\on\default\volume`) and `/X:/non/default/volume` (or `/X:\non\default\volume`)
where I don't think I can just get rid of the prefix, unless I'm missing something.

Reply:
I ran these through a test program:

```
func getURLPath(urlString string) (string, error) {
	url, err := url.Parse(urlString)
	if err != nil {
		return "", err
	}
	path := url.Path
	path = strings.TrimPrefix(path, "/")
	path = filepath.Clean(path)
	return path, nil
}

func main() {
	for _, s := range []string{
		`binary:///path/on/default/volume`,
		`binary:///path\on\default\volume`,
		`binary:///X:/non/default/volume`,
		`binary:///X:\non\default\volume`,
	} {
		path, err := getURLPath(s)
		if err != nil {
			path = fmt.Sprintf("error(%s)", err)
		}
		fmt.Printf("%s => %s\n", s, path)
	}
}
```

This is the output:

```
binary:///path/on/default/volume => path\on\default\volume
binary:///path\on\default\volume => path\on\default\volume
binary:///X:/non/default/volume => X:\non\default\volume
binary:///X:\non\default\volume => X:\non\default\volume
```

It looks like the `X:` cases handle just fine. For the first two cases, I think we would really want them to parse as `\path\on\default\volume` (with a leading slash). So we may need to do something like conditionally only do `TrimPrefix` if the string has a drive letter. We should try to find an elegant way to handle this if possible.

Reply:
On taking another look, I think what you have now will work fine, but we should be able to remove the `strings.Replace` call above. `filepath.Clean` will already convert `/` to `\`.